Use post build plugin

ghubstan opened the pull request:
Imports pre-compiled script plugin `bisq.post-build` instead of custom task `postBuild`.

Based on https://github.com/bisq-network/bisq-pricenode/pull/6

cbeams commented:
@ManfredKarrer, please have a look at commit 5a67f54395e8ed23805d8ad9e4ce4fb9aa07da7f, which takes us off the paid BitcoinAverage developer plan and moves us onto a free endpoint in their API with no change in the data our pricenodes produce. See the full commit comment for details.

There's no need for you to review everything here yet (I'll still have a few more commits coming), but it would be good to get a sanity check on this particular change as soon as possible. I see no reason not to do it; perhaps this endpoint wasn't there when we originally coded this implementation, or perhaps we just didn't know about it? In any case, it simplifies things nicely and reduces costs at the same time.

ManfredKarrer commented:
@cbeams Regarding the ```short``` endpoint: 
Earlier we used ask and bid to have different prices for buy and sell but that was removed a while ago as it confused users. So if there are no issues with request limits we could use that. Best to test it on one node first.

cbeams commented:
I left my local development node running overnight and woke up to errors like the following:

> Rate limit exceeded for unauthenticated requests. You are allowed to make 700 requests every 1440.0 minutes

That limit appears to be undocumented, and it appears to be across all unauthenticated requests from a given IP, not just a specific local/global short ticker endpoint. So that means no more than 1 request every 2 minutes (700/1440 = 0.49), across both our local and global requests. That would work out to 1 request every 4 minutes for each of local and global, which is not great.

Worse, though, is that on a second look at their [plans](https://bitcoinaverage.com/en/plans), non-authenticated use is limited to 5,000 requests per month, which takes the 0.49 number above and reduces it to 0.11 requests per minute. If that's actually enforced, we'd only be able to update local and global data once every 20 minutes.

I'll revert this change, and we can stay with the paid plans.

ManfredKarrer commented:
@cbeams Yes I suspected that they have a limit, should be documented though.... I also like to support them financially, they do a great service and the costs are reasonable. Developer plan is sufficient for one operator.

cbeams commented:
@bisq-network/pricenode-operators, this PR is quite close to complete, and I'd appreciate it if each of you would sync up to it on your local dev machine, read through the README, follow the instructions, make sure everything makes sense to you, and provide any feedback.

@ManfredKarrer is planning to do a proper review of the changes next week, but @mrosseel and @Emzy, any feedback you have in the meantime is welcome.

Note that I'll be taking over one of Manfred's pricenode onion addresses on Monday, and I'll be running it from the HEAD of this PR branch. I'll let it run for a couple of days to fully test it, and then (assuming all is well with Manfred's review), we'll merge this branch to master, and then have everyone update to `v0.7.1` (which will be the new version).

cbeams commented:
Per https://github.com/bisq-network/exchange/pull/1372, I am now operating @ManfredKarrer's old 5bmpx76qllutpcyp.onion pricenode. I'll leave this running for the next 48 hours or so, and when @ManfredKarrer is done with his review (whichever comes last), I'll merge this branch and release v0.7.1, at which point all other @bisq-network/pricenode-operators can upgrade to it.

cbeams commented:
Update: my pricenode built from this PR branch has been running in production for ~48 hours now with no problems. I'm averaging around 35 Bisq clients querying my node during any 1-2 hour period, including my own Bisq client, which is configured to use my pricenode exclusively.

There have been no errors that I am aware of during this period (certainly no failures in the app or stacktraces), and the instance has been stable with regard to resource consumption.

@ManfredKarrer has a lot on his plate right now with getting 0.6.6 out the door for next week's USD liquidity push, so I'm proposing here to all @bisq-network/pricenode-operators that I go ahead and merge this PR without formal review, in light of the fact that the changes clearly work. This will allow for tagging the release as `v0.7.1`, getting all other nodes upgraded, and we can then move forward with further incremental changes. 

Note that I plan to do a YouTube "tech session" on this newly-refactored pricenode and our overall price information infrastructure, so that can fulfill some of the goals of a review, at least as far as information sharing is concerned.

So: @Emzy, @mrosseel, @ManfredKarrer, please respond with an ACK here if you're OK to merge as-is. If you're not OK with that and want to see a proper review beforehand, please say so as well, and also please prioritize performing that review so we can get these changes live. Thanks!

Emzy commented:
ACK

ManfredKarrer commented:
I will do the review now. Sorry for the delay...

cbeams commented:
No worries, @manfredkarrer—and thanks!

ManfredKarrer commented on src/main/java/bisq/price/mining/FeeRate.java:
I assume the reason for calling it price is to see the fee estimation service as "price for the tx fee" service. I still find it not very intuitive to call it price. Conceptually market prices and fee estimation should be different projects, but we do not want to connect to 2 services to avoid network overhead and usability drawbacks as Tor is slow.

ManfredKarrer commented:
Other comments:
The movingAverage was probably a bad idea as with rapid changes we have not adopted fast enough, that's why we have set the parameter to 1 deactivating it in fact. From my point of view we can remove that functionality (and remove the capacity parameter). 

The min value might be decreased as we see atm fee < 10 sat/bytes. Though we should not go too low as if the service fails and provide too low numbers we still play safe and fees < 10 sat/byte are already low enough to be not a concern. 
```private static final long MIN_FEE_RATE = 10; // satoshi/byte```

ManfredKarrer commented on src/main/java/bisq/price/mining/providers/BitcoinFeeRateProvider.java:
In the old version there was no default value set if the service does not provide data. The application itself used a default fee of 200 sat/byte as long the service has not responded. There is currently not handled the case that the service responds but the BTC entry is not available. I think we should do the handling for such case in the application and not in the service as it can be anyway that the connection to the price service fails and thus we need to handle such cases anyway in the Bisq app.

ManfredKarrer commented on src/main/java/bisq/price/mining/providers/BitcoinFeeRateProvider.java:
The current nodes use those values:
DEFAULT_CAPACITY = 1;
DEFAULT_TTL = 2;

ManfredKarrer commented on src/main/java/bisq/price/PriceProvider.java:
For me ttl is confusing as I would expect that it is used for marking data invalid after that time but it is used here  for the request interval.

cbeams commented:
> I do not find the implementation of the previously called method removeOutdatedPrices.
> That was to avoid that the service reports prices in case the provider is offline for longer than 30 min. We prefer to do not show a price in such cases and avoid incorrect prices (in case of % based prices that would lead to wrong trade prices).

The Git log "pickaxe" (`git log -S`) can be most helpful for tracking down why something like this changed. Check out the result of `git log -S removeOutdatedPrices`:

```
$ git log -S removeOutdatedPrices
commit f7927f91e703dd3e52e4c8a6bb02d7bc29c19cc6
Author: Chris Beams <chris@beams.io>
Date:   Fri Feb 9 10:46:56 2018 +0100

    Remove ExchangeRateService#removeOutdatedPrices

    This is no longer necessary now that the data structure that stores this
    information is no longer long-lived. It is recreated on each call to
    getAllMarketData, which in turn returns the latest cached data from
    underlying providers. This data is never older than the individual
    providers' TTLs, so there is never 30-minute old data to purge, which is
    what this method was for; it's just obsolete now.
```

So, as you can see, I had a good reason for eliminating the method, but it was based on a wrong assumption about its purpose. I thought about asking you about that one, actually—should have listened to my intuition! I'll work the functionality back in a self-documenting way. Thanks!

cbeams commented:
Note above that it's a coincidence that the log message contains the string `removeOutdatedPrices`—that's not what the `git log -S removeOutdatedPrices` is matching on. What it matches on is diffs in the codebase. With `-S <string>`, you're asking git to _find the last commits in which `<string>` was added or removed from the codebase`. In this case, it takes us directly to the commit that _removed_ the method. It's worth looking at the whole patch actually, with `git log -p -S removeOutdatedPrices` to see what I mean:
```diff
commit f7927f91e703dd3e52e4c8a6bb02d7bc29c19cc6
Author: Chris Beams <chris@beams.io>
Date:   Fri Feb 9 10:46:56 2018 +0100

    Remove ExchangeRateService#removeOutdatedPrices

    This is no longer necessary now that the data structure that stores this
    information is no longer long-lived. It is recreated on each call to
    getAllMarketData, which in turn returns the latest cached data from
    underlying providers. This data is never older than the individual
    providers' TTLs, so there is never 30-minute old data to purge, which is
    what this method was for; it's just obsolete now.

diff --git src/main/java/bisq/price/spot/ExchangeRateService.java src/main/java/bisq/price/spot/ExchangeRateService.java
index 0df76a6..afb0a10 100644
--- src/main/java/bisq/price/spot/ExchangeRateService.java
+++ src/main/java/bisq/price/spot/ExchangeRateService.java
@@ -20,8 +20,6 @@ package bisq.price.spot;
 import bisq.price.spot.providers.BitcoinAverage;
 import bisq.price.spot.support.CachingExchangeRateProvider;

-import java.time.Instant;
-
 import java.io.IOException;

 import java.util.Collection;
@@ -29,12 +27,9 @@ import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;

 public class ExchangeRateService {

-    private static final long MARKET_PRICE_TTL_SEC = 1800; // 30 min
-
     private final List<ExchangeRateProvider> providers;

     public ExchangeRateService(List<ExchangeRateProvider> providers) {
@@ -83,7 +78,7 @@ public class ExchangeRateService {
             allData.putAll(provider.request());
         }

-        allMarketPrices.put("data", removeOutdatedPrices(allData).values().toArray());
+        allMarketPrices.put("data", allData.values().toArray());
     }

     private long findFirstTimestampForProvider(Collection<ExchangeRateData> prices, String providerSymbol) {
@@ -93,12 +88,4 @@ public class ExchangeRateService {
             .orElseThrow(() -> new IllegalStateException("No exchange rate data found for " + providerSymbol))
             .getTimestampSec();
     }
-
-    private Map<String, ExchangeRateData> removeOutdatedPrices(Map<String, ExchangeRateData> map) {
-        long now = Instant.now().getEpochSecond();
-        long limit = now - MARKET_PRICE_TTL_SEC;
-        return map.entrySet().stream()
-            .filter(e -> e.getValue().getTimestampSec() > limit)
-            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-    }
 }
```

cbeams commented:
> I also do not see the implementation of the logic that we only add the global price if the local price is not available.
> ```
>  		// we don't replace prices which we got form the local request, just in case the global data are received
>         // earlier at startup we allow them but the local request will overwrite them.
>         map.entrySet().stream()
>                 .filter(e -> btcAverageLocalMap == null || !btcAverageLocalMap.containsKey(e.getKey()))
>                 .forEach(e -> allPricesMap.put(e.getKey(), e.getValue()));
> ```

Good catch. This is accounted for, but is being handled in a different way. First, each of our four `ExchangeRateProvider` implementations has an explicit order as seen here:
```
$ git grep -h -A1 @Order
@Order(1)
public static class Global extends BitcoinAverage {
--
@Order(2)
public static class Local extends BitcoinAverage {
--
@Order(3)
class CoinMarketCap extends ExchangeRateProvider {
--
@Order(4)
class Poloniex extends ExchangeRateProvider {
```

At injection time, Spring ensures that any `List<ExchangeRateProvider>` that needs to be injected will contain an instance of each, and will be ordered according to the `@Order` value of each instance. This is what we get as a constructor parameter to `ExchangeRateService`:

```java
    public ExchangeRateService(List<ExchangeRateProvider> providers) {
        this.providers = providers;
    }
```

From there, in `ExchangeRateService#getAllMarketPrices` we iterate through these ordered `providers`, populating the `allExchangeRates` (keyed by currency code) and we use the replace-if-present semantics of `Map#put` operations to ensure that the last provider to report a price for a given currency "wins" and overwrites that entry in the map.

The comment in the original code read as follows:
```java
// we don't replace prices which we got form the local request, just in case the
// global data are received earlier at startup we allow them but the local request
// will overwrite them.
```

Meaning, as I took it, that **local values should always override global values.** And that's why we see `BitcoinAverage$Global` having `@Order(1)` and `BitcoinAverage$Local having` `@Order(2)` above: because local prices should take precedence.

> [..]
> I also don't see the implementation of the logic that we only use Coinmarketcap prices in case Poloniex does not deliver a price.

Same story here. `CoinMarketCap` has `@Order(3)` and `Poloniex` has `@Order(4)`, and therefore `Poloniex`-provided `ExchangeRate` values win.

This approach could be better documented, no doubt. I did actually originally document it in 96a98fd9b423a141af69eac052e63f2e04bc8174, but that was with a previous, pre-Spring migration implementation.

- [ ] @cbeams: Add Javadoc to `ExchangeRateService` constructor and at the `ExchangeRateProvider` class level mentioning the semantics of `@Order` annotations on provider implementations.

cbeams commented on README.md:
- [ ] @cbeams: Prune dead Doge code in a fashion friendly to a future `git revert`

cbeams commented on src/main/java/bisq/price/PriceController.java:
Yeah. It's because I've left Lombok out of the new project altogether. We briefly discussed this when I initially extracted the pricenode module here into its own repository. But it bears treating again here:

I was the one who originally brought Lombok into the Bisq codebase in its early days. I had been—and in many ways still am—a fan of Lombok annotations. They work wonders for getting rid of line noise, especially with regard to eliminating the need to manually code getters and setters and eliminating the need to generate and maintain `hashCode`, `equals` and `toString` methods. It also gets rid of the need to specify a `Logger` for every class. Lombok is wonderful for all those reasons and more.

The problem is that it violates the _principle of least surprise_ for the majority of Java developers that come across Lombok-annotated classes. They see a strange beast, adorned with `@Data`, or `@Value` or `@EqualsAndHashCode` and/or, indeed `@Slf4j` annotations. They see no getters or setters declared within these classes, but then do see getters and setters being _called_ on their instances. Slowly, and probably painfully, they realize what Lombok is and what it's doing: it's generating these elements at the bytecode level using a Java Agent-based annotation processor.

Sometimes people are forced abruptly into figuring this out because they try to import Bisq into IDEA or elsewhere, and they see compilation errors. Then they realize, painfully, that they need a Lombok _plugin_ for their IDE. They find and install it, they rebuild and everything is OK. But it was frustrating getting there.

I was a fairly big champion of Lombok coming from the world of mostly closed-source enterprise projects, where it's a reasonable tradeoff to use something like Lombok widely within your organization, to document it in your new employee onboarding docs and be done with it. The cost of getting a closed, well-paid team on the same page about this is low, and the benefits fairly high.

I think that cost/benefit ratio changes with a project like Bisq. I think it's important that Bisq's components be as simple, understandable and idiomatic as possible. I want any highly experienced Java developer to take one look at a repository like this and feel right at home, instantly able to reason about what's going on, instantly able to be productive. To me, that means using stock-standard Java. It means using widely accepted naming conventions and patterns like those found in Domain Driven Design. It means using well-known, well-supported, thriving open source infrastructure like Spring Boot and the rest of the Spring family provide. It means striving for a situation where any competent Java developer can clone any of our repositories, run a single build command or import a project directly into their IDE of choice and have everything "just work".

Keeping Lombok around in new Bisq repositories just doesn't live up to this standard for me. Lombok is something that's great for a closed, fixed-size team, but is a barrier to entry if you're trying to build an open, global, amorphous team like the Bisq DAO. We want to make the process of contribution as smooth as possible for everyone—especially new people. I just don't see the convenience Lombok offers as being worth it anymore with these goals in mind.

P.S. Other issues can crop up with Lombok, too, especially where new language-level Java features are involved. I can't cite specific examples off the top of my head, but more than once, I've had to deal with Lombok being unable to modify or otherwise deal with certain bytecode. I remember problems coming up around Java 5 as well as around Java 8 if memory serves. We want to adopt Java 9 here as soon as possible, and the fewer annotation processors and agents, etc we have is probably for the better.

cbeams commented:
> The movingAverage was probably a bad idea as with rapid changes we have not adopted fast enough, that's why we have set the parameter to 1 deactivating it in fact. From my point of view we can remove that functionality (and remove the capacity parameter).

Got it, thanks. I wondered about this.

- [ ] @cbeams: Remove `movingAverage` logic and `capacity` parameter
- [ ] @mrosseel: stop checking for capacity value from `/getParams` response (actually, you can just stop calling `/getParams` altogether—I want us to stop coordinating nameless parameter changes via Slack, and instead simply keeping our instances running the latest HEAD of `bisq-network/pricenode:master`, where future such parameter tweaks will take place as commits)

mrosseel commented:
@cbeams will stop the /getParams checking once 0.7.1 is released

cbeams commented:
> The min value might be decreased as we see atm fee < 10 sat/bytes. Though we should not go too low as if the service fails and provide too low numbers we still play safe and fees < 10 sat/byte are already low enough to be not a concern.
> ```java
> private static final long MIN_FEE_RATE = 10; // satoshi/byte
> ```

Yeah, I think I'll leave it as-is, until we've seen more than a few consecutive weeks of these super low sat/byte fees. Also, I'd prefer not to make this change here in this PR in the spirit of keeping thes changes refactoring-based and not about changing behavior.

cbeams commented on src/main/java/bisq/price/mining/FeeRate.java:
What we used to call "market prices" in the code are now called, in their fully-qualified form, _spot exchange rate prices_. We see this reflected in the package, class, and member naming of `bisq.price.spot.ExchangeRate#price`. From https://www.investopedia.com/terms/s/spotprice.asp:

> **What is the 'Spot Price'**
> A spot price is the current price in the marketplace at which a given asset such as a security, commodity or currency can be bought or sold for immediate delivery. While spot prices are specific to both time and place, in a global economy the spot price of most securities or commodities tends to be fairly uniform worldwide. In contrast to spot price, a security, commodity or currency's futures price is its expected value at a specified future time and place.

And from https://www.investopedia.com/terms/s/spotexchangerate.asp:
> **What is a 'Spot Exchange Rate'**
> A spot exchange rate is the price to exchange one currency for another for immediate delivery. The spot rates represent the prices buyers pay in one currency to purchase a second one. Although the spot exchange rate is for delivery on the earliest value date, the standard settlement date for most spot transactions is two business days after the transaction date.

What we used to call "fee estimations" in the code are now called, in their fully-qualified form, _mining fee rate prices_. We see this reflected in the naming of `bisq.price.mining.FeeRate#price`. We use fee _rate_ because that's what satoshis/byte is: a rate at which you must pay a fee, not the actual, total fee.

The specific value of these two rates is known as its _price_, i.e. 

 - "13 satoshis per byte is the current **price** you must pay to get a Bitcoin transaction mined within 1 block.
 -  "The current **price** for 1 bitcoin is 10,314 US dollars"

So we used to call things either "market prices" or "fee estimations". Now we talk about two different _kinds_ of prices. Exchange rate prices and mining fee rate prices. This normalizes things and allows for introducing new prices down the road, like _futures_ prices, etc.

cbeams commented on src/main/java/bisq/price/mining/providers/BitcoinFeeRateProvider.java:
In my analysis, as I recall it anyway, the earlier implementation _did_ return `MIN_FEE_RATE` in the case that no BTC price was available. Perhaps I was mistaken. I'll take another look.

- [ ] @cbeams: Revisit this, decide what to do.

cbeams commented on src/main/java/bisq/price/mining/providers/BitcoinFeeRateProvider.java:
Got it, thanks. I did see that in the monitoring, should have updated it already. I asked about this in Slack, and I got that everybody was running `2 5 3` in production. Either that was bad info, or there was another communication afterward to change these params to `1 5 3` after I asked about it. In any case, it proves my point that, in the future, after this refactoring is merged, we should propagate these parameter changes as _commits_, not as one-off parameter changes. That allows us to communicate things rigorously through Git, not through manual, one-off Slack coordination.

Basically, the protocol for pricenode (and eventually seednode, etc) operators should become something like the following:

 - Watch the bisq-network/pricenode repository, so you always get notified of new commits to its `master` branch
 - Make sure your node is up to date with what's on bisq-network/pricenode:master.

This process should require some human intervention, but it should be as minimal as possible, e.g. syncing your fork's master to upstream/master. This manual step ensures that operators knowingly _choose_ every upgrade, but the step should be as minimal and quick to do as possible, to ensure that this kind of maintenance isn't a burden.

cbeams commented on src/main/java/bisq/price/PriceProvider.java:
I think I agree.

- [ ] @cbeams: Consider changing `ttl` to `interval`, as in "the interval with which this `PriceProvider`'s `refresh()` method is called by its `timer`.

cbeams commented:
Thanks, @ManfredKarrer for the thoughtful review. I believe I've addressed all your comments thus far, and I've added todo checkboxes for myself wherever there's something I need to follow up on.

I should have mentioned earlier that you might want to look at `TODO.md` if you didn't already. This was my running list throughout the refactoring, and I still intend to take care of what remains there. I'll transfer them to proper GitHub issues when we merge this PR, as everything there can and should be done outside the context of this refactoring. In any case, I thought you (and others) might like to see what's there to get a sense of some of the next things I have in mind.

One thing, though, that I do want to get in from the `TODO.md` before merging, is:

 - [ ] @cbeams: Add integration tests

ManfredKarrer commented on src/main/java/bisq/price/PriceController.java:
I agree to most points and I also prefer to stick with plain java instead of too much magical frameworks where someone who is not familiar with need to first learn about it.
Basically the 2 main exceptions for me was Guice and Lombok just because both add that much value to not need boilerplate code that I think its worth it. For the price project I don't care so much as there are not a ton of use cases for Lombok but in the Bisq app it would re-introduce a lot of boilerplate code. Furthermore Lombok is pretty simple to understand (how to use it - not how it works behind the scenes). 
But I agree to the concerns regarding Java version update issues.
Furthermore I prefer also to keep that magic and framework usage limited for security reasons. It is easier to add a vulnerability from a 3rd party framework (as we had in the past with Java serialization - there are Apache commons classes which are vulnerable for hijacks, we did not use those classes but who know if another 3rd party lib uses it...).

cbeams commented on src/main/java/bisq/price/PriceController.java:
Got that, thanks. And to be clear, I am not suggesting that we back Lombok out of the existing bisq-network/exchange repository. This is just where new stuff (or newly extracted) stuff is involved.

ManfredKarrer commented:
Re removeOutdatedPrices:
In case a provider is not responding (happens from time to time) for longer then 30 min. we would get delivered outdated prices as it is. 
The code in the app is handling the caching and the age of the price as well so out dated prices will get ignored there. So from that point of view it it probably ok as it is now.
I perfer to have security related aspects in the app to reduce required trust into the price nodes.


Re provider priotities:
I assumed as well that the order annotation is the key for that ;-). 
But I am worried that this is not very clear, specially for those who are not familiar with the Spring framework.
I would prefer to have that important logic more explicit.


Re TODO:
Agree with most...
- I find the pretty-printing json better if I check the data on Tor browser, not suere if bandwith costs are really so high?
- Re dependency to CurrencyUtil: Do you have a good idea for that? The list changes with each release when we add altcoins.


A general though about the price node is that it is in reality a data aggregation node. We might add maybe more data in future there like historucal trade data which are delivered by the seed node now.
I don't see it as kind of micro service as TOR connections are expensive and slow so we try to collect all calls together to have the min. required requests from the Bisq app to the service.
There is headroom for optimizing the current state where we request fee and market price separately. 
The only reason why we added those nodes was because of problems to API providers over Tor (Cloudflare) - initially the Bisq apps requested directly from BTCA, Polo etc... but only BTCA was working over TOR the other required clearnet connections.
With that background I think it is worth considering a renaming of the project to a more generic name and emphasising on the aggregation aspect.

Another future aspect is to use protobuffer for the data transfer to the Bisq app. Not 100% sure about that but once we would use more complex data structures it might become cumbesome to need to translate json data to the native Bisq data models.
Though that would require dependency on the Bisq core app.
We can discuss that later once we have a concrete use case....

For the PR I leave it you you when you want to merge (adding the small changes first or after the PR). So from my side there is a uACK.

cbeams commented on README.md:
Actually, I'm deferring this to a separate issue, as it's not part of refactoring, per se. See https://github.com/bisq-network/pricenode/issues/9.

cbeams commented on src/main/java/bisq/price/mining/providers/BitcoinFeeRateProvider.java:
- [x] @cbeams: I fixed this in https://github.com/bisq-network/pricenode/pull/7/commits/206c398dc974d91ce47d0fcbb2fa22875d724764

cbeams commented on src/main/java/bisq/price/PriceProvider.java:
Fixed in https://github.com/bisq-network/pricenode/pull/7/commits/3df25dda0fe77888e84c13d96583c6df78a3e071

cbeams commented on src/main/java/bisq/price/mining/providers/BitcoinFeeRateProvider.java:
Leaving this as is, per Manfred's comments that the application should (and does) handle this case.

cbeams commented:
Ok, this PR has (at long last) been merged. I'm holding off on tagging v0.7.1 until we cut releases of underlying bisq-* dependencies, which will happen soon as part of the forthcoming bisq-desktop v0.7.0 release.

In the meantime, it would be good if all @bisq-network/pricenode-operators could at least do a test deployment of commit 545887c, to make sure you understand everything, and that we root out any issues before tagging v0.7.1.

ghubstan committed:
Make test logging configurable

ghubstan committed:
Add submodule bisq-gradle

ghubstan committed:
Remove google.osdetector plugin dependency

ghubstan committed:
Upgrade to Gradle 7.5

Fixes v 7.4.2 bug:  some misleading warnings about task dependencies

ghubstan committed:
Use postBuild task in .gitmodule bisq-gradle

TODO Create new PR to change .gitmodules content

    [submodule "bisq-gradle"]
	    path = bisq-gradle
	    url = https://github.com/ghubstan/bisq-gradle.git
	    branch = add-post-build-task

TO

    [submodule "bisq-gradle"]
	    path = bisq-gradle
	    url = https://github.com/bisq-network/bisq-gradle.git

We need to wait until PR https://github.com/bisq-network/bisq-gradle/pull/1
to be approved & merged before making the above change to .gitmdoules

ghubstan committed:
Adjust paths to new bisq-pricenode repo

TODO Test service installation again after PR merge.
TODO Test scripts with bisq-monitor service dependency
     Wating for merge (new PR) for branch
     https://github.com/ghubstan/bisq-monitor/tree/add-gradle-task-submodule

ghubstan committed:
Formatting & style adjustments

ghubstan committed:
Formatting & style adjustments

ghubstan committed:
Use plugin task

ghubstan committed:
Remove submodule bisq-gradle

ghubstan committed:
Re-add submodule bisq-gradle

ghubstan committed:
Remove my fork's submodule 'bisq-gradle'

Is being replaced by bisq-network's bisq-gradle plugin because PR
https://github.com/bisq-network/bisq-gradle/pull/2 has been merged.

ghubstan committed:
Re add submodule bisq-network bisq-gradle

ghubstan committed:
Re-enable BTCMarkets test

Seems to be blocked only for me, in my region

ripcurlx reviewed (approved):
utACK
